vmm/src/lib.rs

```diff
@@ -1422,6 +1422,15 @@ impl Vmm {
                     VmConfigError::InvalidVcpuCount,
                 ));
             }
+            #[cfg(target_arch = "x86_64")]
+            {
+                if vcpu_count_value as u32 >= x86_64::MAX_SUPPORTED_CPUS {
```

Comment:
Is there a reason why we shouldn't use `KVM_CAP_MAX_VCPUS`? That should theoretically provide a better number per system. We already have a wrapper over this capability in the kvm crate: `get_max_vcpus`.

Reply:
That KVM_CAP_MAX_VCPUS is not reliable for our use case. The maximum CPUs supported depends on the local APIC working mode. The APIC ID is 8 bit In APIC/xAPIC mode, and 32 bit in x2APIC mode. Currently the KVM_CAP_MAX_VCPUS doesn't check local APIC working mode and always return a const 288. So KVM_CAP_MAX_VCPUS is useless for our case.

Reply:
Don't know about 50/70 rules and use 80 columns before. Will fix it.

Reply:
Thanks. I feel 32 or 64 should be enough for firecracker's usage cases, is that OK for your usage cases?

Reply:
From my point of view 32 vcpus should be enough for the Firecracker usecase. I don't see a case where you want to have more. We can update it later if we discover it is not enough.

Reply:
Thanks, so 32 is ok for now. 
Per #279 , it's a little complex. There may be two sematics about "available" CPUs:
1) available CPUs in the system. libc.sysconf(_SC_NPROCESSORS_ONLN).
2) CPUs available the current process, which may be affected by cpuset and cgroups. The num_cpus crate returns such info.
So what's the preferred sematic? Is it ok to import crate num_cpus?

Reply:
I have revised the commit message according to the 50/72 rule.